### PR TITLE
Add Private Azure Cloud support

### PR DESCRIPTION
**Pull request by 0lvin**

### CODE REVIEW

File: cloudify_azure/auth/oauth2.py [hunk 1]
```diff
@@ -89,15 +91,15 @@ def request_access_token(self):
             'client_id': self.credentials.client_id,
             'client_secret': self.credentials.client_secret,
             'grant_type': constants.OAUTH2_GRANT_TYPE,
-            'resource': constants.OAUTH2_MGMT_RESOURCE
+            'resource': self.credentials.endpoint_resource
```
Comment (EarthmanT) [inline cloudify_azure/auth/oauth2.py]: Is this Azure Stack specific?

File: cloudify_azure/connection.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def request(self, **kwargs):
 
            The *url* parameter should be passed in as a root path
            starting AFTER the base endpoint
-           https://management.azure.com/subscriptions/{subscription-id}.
+           https://management.core.windows.net/subscriptions/{subscription-id}.
```
Comment (EarthmanT) [inline cloudify_azure/connection.py]: Is this Azure Stack specific?

File: cloudify_azure/connection.py [hunk 3]
```diff
@@ -92,10 +92,13 @@ def request(self, **kwargs):
         if url.startswith('/'):
             # Add the endpoint and subscription ID
             url = '{0}/subscriptions/{1}{2}'.format(
-                constants.CONN_API_ENDPOINT,
+                creds.endpoints_resource_manager,
```
Comment (EarthmanT) [inline cloudify_azure/connection.py]: Is this Azure Stack specific?

File: cloudify_azure/tests/resources/test_base.py [hunk 4]
```diff
@@ -44,7 +44,11 @@ def setUp(self):
                 'tenant_id': self.tenant_id,
                 'client_id': '1234567',
                 'client_secret': '1234-1234-1234-1234',
-                'subscription_id': self.subscription_id
+                'subscription_id': self.subscription_id,
+                'endpoint_resource': constants.OAUTH2_MGMT_RESOURCE,
+                'endpoint_verify': True,
+                'endpoints_resource_manager': constants.CONN_API_ENDPOINT,
+                'endpoints_active_directory': constants.OAUTH2_ENDPOINT
```
Comment (EarthmanT) [inline cloudify_azure/tests/resources/test_base.py]: Are these parameters ignored if the user does not provide them?

File: examples/resource_manager/deployment-yaml.yaml [hunk 5]
```diff
@@ -152,13 +153,34 @@ inputs:
   image_sku:
     default: 16.04.0-LTS
 
+  endpoint_resource:
+    default: https://management.core.windows.net/
+
+  endpoints_resource_manager:
+    default: https://management.azure.com
```
Comment (EarthmanT) [inline examples/resource_manager/deployment-yaml.yaml]: If someone uses Azure Stack, this is what they provide? What if they use Azure (no Azure Stack)?

File: cloudify_azure/auth/oauth2.py (hunk 1 above)
Comment (0lvin) [reply cloudify_azure/auth/oauth2.py]: Yes, AzureStack can use different url for auth.

File: cloudify_azure/connection.py (hunk 3 above)
Comment (0lvin) [reply cloudify_azure/connection.py]: Yes, AzureStack can use different url for resource managment.

File: cloudify_azure/tests/resources/test_base.py (hunk 4 above)
Comment (0lvin) [reply cloudify_azure/tests/resources/test_base.py]: Yes, default values is provided in plugins.yaml.

File: examples/resource_manager/deployment-yaml.yaml (hunk 5 above)
Comment (0lvin) [reply examples/resource_manager/deployment-yaml.yaml]: Default values is fully same as used before, and moved from constants to plugin.yaml

**Pull request merged**